option.go

```diff
@@ -113,6 +113,24 @@ func IndentSequence(indent bool) EncodeOption {
 	}
 }
 
+// QuoteStyle specifies how strings should be quoted
+type QuoteStyle int
+
+const (
+	// QuoteStyleDouble causes strings to be quoted with a double quote, "
+	QuoteStyleDouble QuoteStyle = iota
+	// QuoteStyleSingle causes strings to be quoted with a single quote, '
+	QuoteStyleSingle
+)
+
+// QuoteStyle specifies how strings should be quoted
+func QuoteStyleOpt(qs QuoteStyle) EncodeOption {
```

Comment:
What do you think `WithQuote(QuoteStyle) EncodeOption` ( or `Quote(QuoteStyle) EncodeOption` ) as name candidate ?